### ISSUE TITLE
Gradient Clipping w/ Multiple Optimizers

### ISSUE THREAD

**Issue by gibiansky** [PyTorchLightning/pytorch-lightning#4667]
### Before asking: 
No docs helped.

#### What is your question?

I have a GAN with two optimizers, one generator one discriminator. I would like to clip *only* generator parameters.

What is the cleanest way to accomplish this?

Currently just using standard training_step and configure_optimizers with automatic optimization enabled.

**Comment by ananyahjha93** [comment]
@tchaton do you have a suggestion here?

**Comment by gibiansky** [comment]
I implemented it as follows:

```python
    def training_step(self, batch, _batch_idx, _optimizer_idx):
        generator_opt, discriminator_opt = self.optimizers()

        # Train generator.
        generator_loss = self.generator_training_step(batch)
        self.manual_backward(generator_loss, generator_opt)
        torch.nn.utils.clip_grad_norm_(self.generator.parameters(), self.grad_norm_clip)

        self.manual_optimizer_step(generator_opt)

        # Train discriminator.
        discriminator_loss = self.discriminator_training_step(batch)
        self.manual_backward(discriminator_loss, discriminator_opt)
        self.manual_optimizer_step(discriminator_opt)
```

While setting `automatic_optimization=False`.

Is this the right approach? If so, this works for now, but perhaps the trainer gradnorm argument could be extended to allow a tuple of equal length to the length of `self.optimizers()`.

**Comment by rohitgr7** [comment]
IMO gradient clipping should be a part of `LightningModule` rather than the `Accelerator`, so that one can override it and make changes as per the use-case.

**Comment by sophiajw** [comment]
hey, I just had the same question and thought I share the solution that I found: 
This is now possible without manual optimization by using [configure-gradient-clipping](https://pytorch-lightning.readthedocs.io/en/1.5.7/common/lightning_module.html#configure-gradient-clipping) of `LightningModule`.
